### PR TITLE
Remove state, lastRun, parentLastRun, rootLastRun on Runner + add lastRunId instead

### PR DESCRIPTION
**Pull request by jreynard-code**

### CODE REVIEW

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt [hunk 1]
```diff
@@ -77,9 +76,19 @@ class RunnerService(
   }
 
   fun deleteInstance(runnerInstance: RunnerInstance) {
-    if (runnerInstance.isRunning())
-        throw CsmClientException(
-            "Can't delete a running runner : ${runnerInstance.getRunnerDataObjet().id}")
+    val runner = runnerInstance.runner
+
+    if (!runner.lastRunId.isNullOrBlank()) {
```
Comment (lalepee) [inline runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt:81]: Can this be moved to a runnerInstance Function instead ?

**Review by lalepee — CHANGES_REQUESTED** [review]

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt [hunk 2]
```diff
@@ -297,5 +279,24 @@ class RunnerService(
       val rbacSecurity = csmRbac.setDefault(this.getRbacSecurity(), role, this.roleDefinition)
       this.setRbacSecurity(rbacSecurity)
     }
+
+    fun delete() {
```
Comment (lalepee) [inline runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt:283]: The delete should be in the runnerService class. runnerService Class do action on runnerInstance Class. runnerInstance should only provide status.

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerApiServiceImpl.kt [hunk 3]
```diff
@@ -62,8 +61,7 @@ internal class RunnerApiServiceImpl(
   override fun deleteRunner(organizationId: String, workspaceId: String, runnerId: String) {
     val runnerService = getRunnerService().inOrganization(organizationId).inWorkspace(workspaceId)
     val runner = runnerService.getInstance(runnerId).userHasPermission(PERMISSION_DELETE)
```
Comment (lalepee) [inline runner/src/main/kotlin/com/cosmotech/runner/service/RunnerApiServiceImpl.kt:63]: var runner should be named runnerInstance

**Review by lalepee — CHANGES_REQUESTED** [review]
Sorry some change needed again

**Review by neomatamune — APPROVED** [review]

**Pull request merged**